packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2727,6 +2740,10 @@ class TextSelectionGestureDetectorBuilder {
     // See https://github.com/flutter/flutter/issues/115130.
     renderEditable.handleSecondaryTapDown(TapDownDetails(globalPosition: details.globalPosition));
     _shouldShowSelectionToolbar = true;
+    _shouldShowSelectionHandles =
+        details.kind == null ||
+        details.kind == PointerDeviceKind.touch ||
+        details.kind == PointerDeviceKind.stylus;
```

Comment:
I double checked this on native Android and indeed the selection handles show when using a stylus. Good call.